tests/test_builtins.py

```diff
@@ -352,6 +352,66 @@ def test_norm_sparse(self):
         term2 = norm(rec0)
         assert np.isclose(term1/term2 - 1, 0.0, rtol=0.0, atol=1e-5)
 
+    def test_sum(self):
+        """
+        Test that sum produces the correct result against numpy
+        """
+        grid = Grid((11, 11))
+        x, y = grid.dimensions
+        t = grid.stepping_dim
+        f = TimeFunction(name='f', grid=grid)
+        f.data.fill(1)
+        assert sum(f) == sumall(f)
+        assert sum(f, dims=None) == sumall(f)
+        assert sum(f, dims=f.dimensions) == sumall(f)
+        assert sumall(f) == np.prod(f.shape)
+
+        fsum = sum(f, dims=x)
+        assert fsum.dimensions == (t, y)
+        assert np.all(fsum.data == 11)
+
+        fsum = sum(f, dims=t)
+        assert not fsum.is_TimeFunction
+        assert fsum.dimensions == (x, y)
+        assert np.all(fsum.data == 2)
+
+        fsum = sum(fsum, dims=x)
+        assert not fsum.is_TimeFunction
+        assert fsum.dimensions == (y,)
+        assert np.all(fsum.data == 2 * 11)
+
+    def test_sum_sparse(self):
+        """
+        Test that sum produces the correct result against numpy
+        """
+        grid = Grid((11, 11))
+        nx = 101
+        nt = 1001
+        s = SparseTimeFunction(name='s', grid=grid, nt=nt, npoint=nx)
+        s.data.fill(1.)
+        s.coordinates.data[:] = np.random.rand(*s.coordinates.data.shape)
```

Comment:
I think we avoided so far to have randomness in tests (?)

Reply:
Is there some way to set a seed on the rand so it generates a consistent array?

Reply:
We avoid *computation* on random values, this is just whatever value never changed.